python/cuml/metrics/regression.pyx

```diff
@@ -140,47 +193,106 @@ def mean_squared_error(y_true, y_pred,
+
+    Returns
+    -------
+    loss : float or ndarray of floats
+        If multioutput is ‘raw_values’, then mean absolute error is returned
+        for each output separately. If multioutput is ‘uniform_average’ or an
+        ndarray of weights, then the weighted average of all output errors is
+        returned.
 
+        MAE output is non-negative floating point. The best value is 0.0.
+    """
+    y_true, y_pred, sample_weight, multioutput, raw_multioutput = \
+        _prepare_input_reg(y_true, y_pred, sample_weight, multioutput)
+
+    output_errors = cp.abs(cp.subtract(y_pred, y_true))
     output_errors = cp.average(output_errors, axis=0, weights=sample_weight)
 
     if raw_multioutput:
         return output_errors
 
-    mse = cp.average(output_errors, weights=multioutput)
-    return mse if squared else cp.sqrt(mse)
+    return cp.average(output_errors, weights=multioutput)
+
+
+@with_cupy_rmm
+def mean_squared_log_error(y_true, y_pred,
+                           sample_weight=None,
+                           multioutput='uniform_average',
+                           squared=True):
+    """Mean squared log error regression loss
+
+    Be careful when using this metric with float32 inputs as the result can be
+    slightly incorrect because of floating point precision if the input is
+    large enough. We recommend to use float64 instead.
+    Parameters
+    ----------
+    y_true : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+    y_pred : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Estimated target values.
+    sample_weight : array-like (device or host) shape = (n_samples,), optional
+        Sample weights.
+    multioutput : string in ['raw_values', 'uniform_average']
+        or array-like of shape (n_outputs)
+        Defines aggregating of multiple output values.
+        Array-like value defines weights used to average errors.
+        'raw_values' :
+            Returns a full set of errors in case of multioutput input.
+        'uniform_average' :
+            Errors of all outputs are averaged with uniform weight.
+    squared : boolean value, optional (default = True)
+        If True returns MSE value, if False returns RMSE value.
+
+    Returns
+    -------
+    loss : float or ndarray of floats
+        A non-negative floating point value (the best value is 0.0), or an
+        array of floating point values, one for each individual target.
+    """
+    y_true, y_pred, sample_weight, multioutput, raw_multioutput = \
+        _prepare_input_reg(y_true, y_pred, sample_weight, multioutput)
+
+    if cp.less(y_true, 0).any() or cp.less(y_pred, 0).any():
+        raise ValueError("Mean Squared Logarithmic Error cannot be used when "
+                         "targets contain negative values.")
+
+    return _mse(cp.log1p(y_true), cp.log1p(y_pred), sample_weight, multioutput,
```

Comment:
Does sklearn use log1p? I think the interface implies it'll just be cp.log

Reply:
Yes they do, here is the formula from sklearn's user guide: https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error I agree with you that the name is confusing though.